src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1316,13 +1316,14 @@ private String getFirstResultSetCommand(
     } else if (catalog.isEmpty()) {
       return "";
     } else {
+      String catalogUnescaped = unescapeChars(catalog);
```

Comment:
We do not support wildcards for catalogs (only schema patterns and other args, like table name pattern or column name pattern), so characters other than the double quotes shouldn't be altered for the catalog name specifically. Documented behavior here: https://docs.snowflake.com/en/user-guide/jdbc-api.html#object-databasemetadata 

Instead of unescapeChars() for the catalog, I made a function escapeSqlQuotes() used in getSchemas() that can be used instead. So `String catalogUnescaped = escapeSqlQuotes(catalog);`